onnxruntime/core/providers/openvino/openvino_graph.cc

```diff
@@ -0,0 +1,369 @@
+		throw "OpenVINO Inference: Inputs count mismatch!";
+	}
+
+	auto graph_output_info = cnn_network_->getOutputsInfo();
+	if (num_outputs != graph_output_info.size()) {
+		throw "OpenVINO Inference: Outputs count mismatch!";
+	}
+
+	//
+	// Copies the same input to all infer request blobs and
+	// starts an async inference on each of them.
+	// Output from only the first infer_request is returned.
+	//
+
+
+
+	// Prepare input
+	for(auto infer_request : infer_requests_) {
+
+    size_t i = 0;
+    for (auto input_info_iter = graph_input_info.begin();
+        input_info_iter != graph_input_info.end(); ++input_info_iter, ++i) {
+
+      // Get OpenVINO's input buffer
+      auto graph_input_blob = infer_request->GetBlob(input_info_iter->first);
+      auto graph_input_buffer =
+          graph_input_blob->buffer().as<
+              InferenceEngine::PrecisionTrait<InferenceEngine::Precision::FP32>::value_type*>();
+      // Get data size
+      size_t num_input_elements = 1;
+      for (auto dim : input_info_iter->second->getTensorDesc().getDims()) {
+        num_input_elements *= dim;
+      }
+
+      size_t input_data_size = num_input_elements * sizeof(float);
+    //   for(int j=0; j< num_input_elements; j++){
+    //       std::cout << "Input tensors " << ((float*)input_tensors[i].data)[j] << std::endl ;
+    //   }
+      // Copy input data into OpenVINO's input buffer
+      std::memcpy(graph_input_buffer, input_tensors[i].data, input_data_size);
+
+    }
+  }
+
+	// Start Async inferences
+	for(auto infer_request : infer_requests_) {
+	  infer_request->StartAsync();
+	}
+
+	// Wait for results
+	for(auto infer_request : infer_requests_) {
+	  infer_request->Wait(InferenceEngine::IInferRequest::WaitMode::RESULT_READY);
+	}
+
+	// Process output
+
+	auto infer_request = infer_requests_[0];
+
+	size_t i = 0;
+	for (auto output_info_iter = graph_output_info.begin();
+			output_info_iter != graph_output_info.end();
+			++output_info_iter, ++i) {
+
+		// Get OpenVINO's output buffer
+		auto graph_output_blob = infer_request->GetBlob(
+				output_info_iter->first);
+		auto graph_output_buffer =
+				graph_output_blob->buffer().as<
+						InferenceEngine::PrecisionTrait<
+								InferenceEngine::Precision::FP32>::value_type*>();
+
+		// Get data size & initialize output tensor info
+		auto graph_output_dims = graph_output_blob->getTensorDesc().getDims();
+        // for(auto dims : graph_output_dims){
+        //     std::cout << "Output Dims are " << dims << std::endl;
+        // }
+
+		auto num_dims = graph_output_dims.size();
+		size_t output_data_size = graph_output_blob->byteSize();
+
+    // TODO: Memory Leak!!!!
+    // fix before shipping.
+		output_tensors[i].shape = new int64_t[num_dims];
+		for (int j = 0; j < num_dims; j++) {
+			output_tensors[i].shape[j] = (int64_t)graph_output_dims[j];
+		}
+
+
+		output_tensors[i].ndim = num_dims;
+		output_tensors[i].dtype = onnxruntime::DType::TFloat32;
```

Comment:
do we not support other types currently?

Reply:
Added support for the following tensor data types for graph inputs and outputs:
    1) float
    2) float16
    3) int32
    4) int16
    5) int8
    6) uint16
    7) uint8